src/oemof/solph/constraints/storage_level.py

```diff
@@ -0,0 +1,177 @@
+
+        setattr(
+            model,
+            constraint_name,
+            po.Constraint(
+                OUTPUTS,
+                model.TIMESTEPS,
+                noruleinit=True,
+            ),
+        )
+        setattr(
+            model,
+            constraint_name + "build",
+            po.BuildAction(rule=_output_active_rule),
+        )
+
+        # Define constraints on the output flows
+        def _constraint_output_rule(m, o, p, t):
+            return (
+                m.flow[multiplexer_bus, o, p, t]
+                / m.flows[multiplexer_bus, o].nominal_value
+                <= active_output[o, t]
+            )
+
+        setattr(
+            model,
+            f"{name}_output_constraint",
+            po.Constraint(
+                OUTPUTS,
+                model.PERIODS,
+                model.TIMESTEPS,
+                rule=_constraint_output_rule,
+            ),
+        )
+
+    _outputs()
+
+    def _inputs():
+        INPUTS = po.Set(initialize=input_levels.keys())
+        setattr(model, f"{name}_INPUTS", INPUTS)
+
+        inactive_input = po.Var(
+            INPUTS, model.TIMESTEPS, domain=po.Binary, bounds=(0, 1)
+        )
+        setattr(model, f"{name}_active_input", inactive_input)
+
+        constraint_name = f"{name}_input_active_constraint"
+
+        def _input_active_rule(m):
+            r"""
+            .. math::
+                \hat{y}_n \ge (E(t) - E_n) - E_{max}
+            """
+            for t in m.TIMESTEPS:
+                for o in input_levels:
+                    getattr(m, constraint_name).add(
+                        (o, t),
+                        (
+                            m.GenericStorageBlock.storage_content[
+                                storage_component, t
+                            ]
+                            - input_levels[o]
+                        )
+                        / storage_component.nominal_storage_capacity
+                        <= inactive_input[o, t],
+                    )
+
+        setattr(
+            model,
+            constraint_name,
+            po.Constraint(
+                INPUTS,
+                model.TIMESTEPS,
+                noruleinit=True,
+            ),
+        )
+        setattr(
+            model,
+            constraint_name + "build",
+            po.BuildAction(rule=_input_active_rule),
+        )
+
+        # Define constraints on the input flows
+        def _constraint_input_rule(m, i, p, t):
+            return (
+                m.flow[i, multiplexer_bus, p, t]
+                / m.flows[i, multiplexer_bus].nominal_value
+                <= 1 - inactive_input[i, t]
+            )
+
+        setattr(
+            model,
+            f"{name}_input_constraint",
+            po.Constraint(
+                INPUTS,
+                model.PERIODS,
+                model.TIMESTEPS,
+                rule=_constraint_input_rule,
+            ),
+        )
+
+    _inputs()
+
+    return
```

Comment:
Can't this be dropped? If I'm not mistaken this is the same as `return None` which has no effect, right?